### PR TITLE
CI macOS: Build following stages even when some job failed; adjust platforms

### PR DESCRIPTION
**Pull request by mkoeppe**
To fix what can be seen in https://github.com/sagemath/sage/actions/runs/6827163283 - stage-2 was not attempted because some of the stage-1 jobs failed.

We also replace one of the failing platforms and remove another.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jhpalmieri** [comment]
What are stages 1 and 2? I have a guess about `stage-2-optional-p-z` or `stage-2-experimental-0-o`, for example, but what are `stage-1` and `stage-2`?

**Comment by mkoeppe** [comment]
Stage 1 builds non-Python packages, stage 2 builds Python packages

File: .github/workflows/ci-macos.yml [hunk 1]
```diff
@@ -40,30 +40,35 @@ jobs:
     with:
       stage: "2"
     needs: [stage-1]
+    if: ${{ success() || failure() }}
```
Comment (jhpalmieri) [inline .github/workflows/ci-macos.yml:43]: Is this equivalent to `if: ${{ always() }}`? (I found that at https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#example-not-requiring-successful-dependent-jobs.) Any reason to prefer one approach to the other?

Comment (mkoeppe) [reply .github/workflows/ci-macos.yml:43]: We just got rid of a lot of `always()` in:
- #36617

**Comment by jhpalmieri** [comment]
Okay, looks good.

**Review by jhpalmieri — APPROVED** [review]

**Comment by mkoeppe** [comment]
Thanks! 

Setting it to blocker so it gets merged in the next rc

**Pull request merged**